subprojects/core/src/main/java/org/gradle/process/internal/worker/request/WorkerAction.java

```diff
@@ -83,7 +88,7 @@ public void execute(WorkerProcessContext workerProcessContext) {
                     // Make the argument serializers available so work implementations can register their own serializers
                     registration.add(RequestArgumentSerializers.class, argumentSerializers);
                     registration.add(InstantiatorFactory.class, instantiatorFactory);
-                    registration.add(InternalProblems.class, new DefaultProblems(ImmutableList.of(new WorkerProblemEmitter(responder)), CurrentBuildOperationRef.instance()));
+                    registration.add(InternalProblems.class, new DefaultProblems(new SummarizerToEmitterAdapter(new WorkerProblemEmitter(responder)), CurrentBuildOperationRef.instance()));
```

Comment:
I don't understand why we need this.

Reply:
related to this.

https://github.com/gradle/gradle/pull/30841/files/8ed7c2276eb2d8c4dc8d13c7e724844e84256ccb#r1827908749

Reply:
The url does not work.